### ISSUE TITLE
Find solution to blocking `eth` calls (during sync)

### ISSUE THREAD

**Issue by farazdagi** [status-im/status-go#102]
**Problem**
- `eth` is not responsive up until sync is done
- tested in `geth --light --testnet console`, no `eth` calls is possible if LES headers sync is in progress

**Notes:**
- this issue is claimed to be solved by https://github.com/ethereum/go-ethereum/pull/3519 (but it isn't)
- issue introduced in https://github.com/ethereum/go-ethereum/commit/c57c54ce96628aeb6345776310123a80593f0143

**Resolution:**
- create bug fix request by describing the issue
- briefly review if solution can be found, if so then file PR to that bug fix request

**Comment by farazdagi** [comment]
Bug fix request submitted: https://github.com/ethereum/go-ethereum/issues/3639

**Comment by jarradh** [comment]
related PR ethereum/go-ethereum#3660

**Comment by farazdagi** [comment]
Note to myself: https://github.com/ethereum/go-ethereum/commit/682875adff760a29a2bb0024190883e4b4dd5d72 has updated gas estimation algorithm, so `eth.estimateGas()` takes slightly more time (measured in 3-5 extra seconds, per my anecdotical evidence)

**Comment by farazdagi** [comment]
We've circumvented the problem by updating all other packages, except for the LES (and we will keep it this way, up until the issue with LES blocking is properly resolved on `go-ethereum` side).

In the meantime, the following branches are available:
- https://github.com/status-im/status-go/commits/geth/1.5.6-stable-with-non-blocking-les
- https://github.com/status-im/status-go/commits/geth/1.5.7-stable-with-non-blocking-les
- https://github.com/status-im/status-go/commits/geth/1.5.8-stable-with-non-blocking-les
- https://github.com/status-im/status-go/commits/geth/1.5.9-stable-with-non-blocking-les

Once https://github.com/status-im/status-go/pull/117 is merged (which is based on https://github.com/status-im/status-go/commits/geth/1.5.9-stable-with-non-blocking-les) it is safe to close this issue.

**Comment by farazdagi** [comment]
- The  https://github.com/status-im/status-go/commits/geth/1.5.9-stable-with-non-blocking-les has been used in `release/0.9.4` - so this issue can be closed.
- Additionally, Zsolt updated his PR which has better responsiveness (yet once again, we will update LES only after I'm **absolutely** sure that no blocking is introduced). That will be tackled as a separate issue: https://github.com/status-im/status-go/issues/121
